### PR TITLE
Accordion more progressive

### PR DESCRIPTION
**Pull request by aaronhans**
This branch rewrites the accordion component so that it uses the details and summary elements which expand and contract in browsers. by default without any javascript.

It still uses javascript in order to apply open and closing animations.

CSS is modified to hide the default open/closing caret indicator and apply our design with a plus/minus icon only when the javascript executes by using :defined on the custom element name in the CSS selector

The plus and minus icons are created with CSS box shadows so the previously required separate components plus and minus are deleted

This is a breaking change for this component because the recommended markup is changed significantly. The major version is bumped to indicate the breaking nature of the update

### CODE REVIEW

File: components/accordion/src/index.scss [hunk 1]
```diff
@@ -1,32 +1,88 @@
...
+  margin: 0;
+  
+  color: var(--primary-color, #064e66);
+  font-size: 1.125rem;
+  font-weight: bold;
+}
+
+cagov-accordion:defined details summary {
+  list-style: none; /* hide default expansion triangle after js executes */
+  border-radius: 5px 5px 0 0;
```
Comment (zakiya) [inline components/accordion/src/index.scss:38]: Do we have a border-radius css variable yet? if not we should open an issue for it.

**Review by zakiya — DISMISSED** [review]
Made some minor suggestions. I also think the designers may want a bit more space between the heading and body. 

CleanShot 2022-02-07 at 09 58 09@2x

File: components/accordion/src/index.scss (hunk 1 above)
Comment (aaronhans) [reply components/accordion/src/index.scss:38]: Thank you!

File: components/accordion/src/index.scss [hunk 2]
```diff
@@ -1,32 +1,88 @@
...
+}
+
+cagov-accordion:defined details summary:focus {
+  outline-offset: -2px;
+  outline: solid 2px var(--highlight-color,#fbad23)!important;
+}
+
+details .accordion-body {
+  padding: 0 0 1rem 1rem;
```
[the remark this answers is not in the document]
Comment (aaronhans) [reply components/accordion/src/index.scss:47]: Thank you!

File: components/accordion/src/index.scss (hunk 1 above)
Comment (aaronhans) [reply components/accordion/src/index.scss]: Using border-radius css var now

File: components/accordion/src/index.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (aaronhans) [reply components/accordion/src/index.scss]: Using many more existing variables now

**Comment by aaronhans** [comment]
Thanks for the tips @zakiya! Padding increased and many more css variables are used

**Comment by zakiya** [comment]
Nicely done!

**Comment by aaronhans** [comment]
This is looking good but the integration into a couple of our sites has not been completed. Ran into some issues with the code generated by the custom Gutenberg block used for the chart drawer on the covid19 site that have to be resolved before we have a release candidate there. 

Discovered several issues during integration already whose fixes have been made in this new version of the component.

Since this is a major release change and site integrations have not been completed by the release deadline this PR will move to the next release candidate

**Comment by aaronhans** [comment]
The code changes in here are good but not going to merge this until the integrations of this new component are fully tested in both:
- covid19.ca.gov: https://github.com/cagov/design-system/pull/685
- and one of the pantheon plugin based WordPress sites, see accordion-v2 branch in ca-design-system-gutenberg-blocks

**Comment by aaronhans** [comment]
Closing this PR, opened new PR against latest release candidate branch here https://github.com/cagov/design-system/pull/698

**Pull request closed without merging**